src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx

```diff
@@ -253,68 +288,96 @@ export const useGetTradeQuotes = () => {
+        skip: !swapperName || !shouldFetchTradeQuotes,
         pollingInterval:
-          swappers[swapperName]?.pollingInterval ?? DEFAULT_GET_TRADE_QUOTE_POLLING_INTERVAL,
+          swappers[swapperName as SwapperName]?.pollingInterval ??
+          DEFAULT_GET_TRADE_QUOTE_POLLING_INTERVAL,
       }
     },
-    [shouldRefetchTradeQuotes, tradeQuoteInput],
+    [shouldFetchTradeQuotes, tradeQuoteInput],
   )
 
-  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.CowSwap))
-  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.ArbitrumBridge))
-  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.Portals))
-  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.LIFI))
-  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.Thorchain))
-  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.Zrx))
-  useGetSwapperTradeQuote(getTradeQuoteArgs(SwapperName.Chainflip))
+  const queryStateMeta = useGetSwapperTradeQuote(
+    getTradeQuoteArgs(activeQuoteMetaRef.current?.swapperName),
+  )
 
   // true if any debounce, input or swapper is fetching
   const isAnyTradeQuoteLoading = useAppSelector(selectIsAnyTradeQuoteLoading)
 
-  const sortedTradeQuotes = useAppSelector(selectSortedTradeQuotes)
-  const activeQuoteMeta = useAppSelector(selectActiveQuoteMetaOrDefault)
-
   // auto-select the best quote once all quotes have arrived
   useEffect(() => {
-    // don't override user selection, don't rug users by auto-selecting while results are incoming
-    if (activeQuoteMeta || isAnyTradeQuoteLoading) return
-
-    const bestQuote: ApiQuote | undefined = selectSortedTradeQuotes(store.getState())[0]
-
-    // don't auto-select nothing, don't auto-select errored quotes
-    if (bestQuote?.quote === undefined || bestQuote.errors.length > 0) {
-      return
-    }
-
-    dispatch(tradeQuoteSlice.actions.setActiveQuote(bestQuote))
-  }, [activeQuoteMeta, isAnyTradeQuoteLoading, dispatch])
+    if (!confirmedTradeExecution) return
+    // We already have an executable active trade, don't rerun this or this will run forever
+    if (activeTrade && isExecutableTradeQuote(activeTrade)) return
+    const identifier = activeQuoteMetaRef.current?.identifier
+    if (!identifier) return
+    if (!queryStateMeta?.data) return
+    const quoteData = queryStateMeta.data[identifier]
+    if (!quoteData?.quote) return
+
+    // Set the execution metadata to that of the previous rate so we can take over
+    dispatch(
+      tradeQuoteSlice.actions.setTradeExecutionMetadata({
+        id: quoteData.quote.id,
+        executionMetadata: confirmedTradeExecution,
+      }),
+    )
+    // Set as both confirmed *and* active
+    dispatch(tradeQuoteSlice.actions.setActiveQuote(quoteData))
+    dispatch(tradeQuoteSlice.actions.setConfirmedQuote(quoteData.quote))
+    // And re-confirm the trade since we're effectively resetting the state machine here
+    // dispatch(tradeQuoteSlice.actions.confirmTrade(quoteData.quote.id))
```

Comment:
Should this be uncommented?

Reply:
Actually not! It's a leftover, nice catch

1982311b89